### PR TITLE
Adding Inventory and host Management plus various tidy up commits

### PR DESCRIPTION
**Pull request by pcuzner**
Multiple changes in this 'bundle', the main features being;
- inventory management: add/remove hosts and groups
- API responses: responses now use a common response object
- Check/create ssh keys on start up
- Add host checks for ssh connectivity, returning our public key if we get "unauthorized" error
- Simplified env/cmdline usage (for tags)

Also addresses the following issues
#21 
#22 
#23

### CODE REVIEW

File: misc/docker/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ store it in the misc/docker directory
  cp {logging,config}.yaml /etc/ansible-runner-service/.
  cp samples/* /usr/share/ansible-runner-service/
```
Comment (jmolmo) [inline misc/docker/README.md]: recursive copy?
cp -r samples/* /usr/share/ansible-runner-service/

Comment (jmolmo) [inline misc/docker/README.md]: I think better to create this folders, or another possibility (even better) is to provide the folders already created in the tar.gz
mkdir /usr/share/ansible-runner-service/artifacts
mkdir /usr/share/ansible-runner-service/env
mkdir /usr/share/ansible-runner-service/inventory

File: misc/docker/README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ store it in the misc/docker directory
...
 ```
-*(the paths don't matter, they're just the ones I'm using for test)*
+*(the paths don't matter, they're just the ones I'm using for test. However, you do **need** artifacts, env, inventory and project within /usr/share/ansible-runner)*
```
Comment (jmolmo) [inline misc/docker/README.md]: If we provide all the folders created, better remove this comment. In any case the comment would be:

(the paths don't matter, they're just the ones I'm using for test, if you change them just take that into account mapping the persistence folders in the command to start the container

File: misc/docker/README.md [hunk 3]
```diff
@@ -50,9 +50,4 @@ docker run -d --network=host -p 5001:5001/tcp --name runner-service runner-servi
 docker run -d --network=host -p 5001:5001/tcp -v /usr/share/ansible-runner-service:/usr/share/ansible-runner-service -v /etc/ansible-runner-service:/etc/ansible-runner-service --name runner-service runner-service
```
Comment (jmolmo) [inline misc/docker/README.md]: I have seen that the container (with persistence) does not start even with the "env" folder created. 
This happens if SELINUX=enforcing (/etc/sysconfig/selinux)
Is needed to change this setting to "permissive/disabled" in order to work.
Another workaround is to "disable" temporally with:
# sudo setenforce 0

File: misc/docker/README.md (hunk 1 above)
Comment (Daniel-Pivonka) [reply misc/docker/README.md]: +1

Comment (pcuzner) [reply misc/docker/README.md]: +1. This is something I've asked Daniel to look at once we merge this PR

File: misc/docker/README.md (hunk 2 above)
Comment (pcuzner) [reply misc/docker/README.md]: README updated in commit e97a9f9

File: misc/docker/README.md (hunk 3 above)
Comment (pcuzner) [reply misc/docker/README.md]: @Daniel-Pivonka @jmolmo Please review the doc and ssh key handling changes added today - I'd like to get this merged in the next couple of days.

Comment (jmolmo) [reply misc/docker/README.md]: It should work perfect now!!  ( and i have learn some things!!)

**Pull request merged**